service/collector/deployment.go

```diff
@@ -90,6 +91,8 @@ func (d *Deployment) Collect(ch chan<- prometheus.Metric) error {
 	}
 	defer watcher.Stop()
 
+	alreadyEmitted := map[string]bool{}
```

Comment:
FYI the more memory efficient version and best practise for these kinds of usages of maps is to use `struct{}` as value.